addon/model.js

```diff
@@ -413,6 +422,7 @@ export default class MegamorphicModel extends EmberObject {
 
     // Remove errors upon setting of new value
     this._removeError(key);
+    flushChanges(this._store);
```

Comment:
Why do we have to flush here if we are not in the push but rather in the `set` flow?

Reply:
the change notification gets deferred in [RecordData.setAttr](https://github.com/hjdivad/ember-m3/blob/hjdivad/batch-change-notfiications/addon/record-data.js#L387)

It seems clearer to me to have a flush here rather than have the change notification be immediate deeper in the system (in record data) but i'm open to being persuaded otherwise